### ISSUE TITLE
想剪枝带pixelshuffle的网络，但是报错了，该怎么适配一下呢？

### ISSUE THREAD

**Issue by ensiwalk** [VainF/Torch-Pruning#98]
`class SimplePixelShuffle(nn.Module):
    def __init__(self,channels):
        super().__init__()
        self.conv1 = nn.Conv2d(in_channels=channels, out_channels=channels, kernel_size=1, padding=0, stride=1,
                               groups=1, bias=True)
        self.pixshuffle = nn.PixelShuffle(2)
        self.conv2 = nn.Conv2d(in_channels=channels//4, out_channels=channels, kernel_size=1, padding=0, stride=1,
                               groups=1, bias=True)
    def forward(self,x):
        x = self.conv1(x)
        x = self.pixshuffle(x)
        x = self.conv2(x)
        return x`

**Comment by AvrahamRaviv** [comment]
Hi,
Is there any progress?

**Comment by Weichongchan666** [comment]
You can try to list the conv2d layers before the pixshuffle layers into ignored_layers through a loop and then prune it.

**Comment by AvrahamRaviv** [comment]
> You can try to list the conv2d layers before the pixshuffle layers into ignored_layers through a loop and then prune it.

Thanks for the answer. I think I didn't understand your meaning, can you explain it please? Thanks!
